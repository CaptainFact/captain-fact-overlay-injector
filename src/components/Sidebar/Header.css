.header {
  background: rgba(0, 0, 0, 0.75);
  padding: 2px 7px 4px;
  font-size: 1.5em;
  border: 1px solid #2f2f2f;
}

.header h1 {
  display: inline-block;
}

.title {
  font-size: 1em;
  margin: 0;
  color: white;
}

.title img {
  height: 0.6em;
  width: 0.6em;
}

.title:hover {
  font-size: 1em;
  text-decoration: underline;
}

.closeBtn {
  float: right;
  color: white;
  cursor: pointer;
}