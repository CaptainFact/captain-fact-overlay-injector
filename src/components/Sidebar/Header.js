import React, { PureComponent } from 'react'

import { FRONTEND_URL } from '../../constants'

import { header, closeBtn, title } from './Header.css'
import DEFAULT_IMG_NEW_TAB from "../../assets/new_tab.png"


export const Header = ({videoHashId, onCloseClick=null, imgNewTab=DEFAULT_IMG_NEW_TAB}) =>
  <div className={header}>
    <a href={`${FRONTEND_URL}/videos/${videoHashId}`} target="_BLANK" title="Open discussion on CaptainFact">
      <h1 className={title}>CaptainFact <img src={imgNewTab}/></h1>
    </a>
    {onCloseClick &&
    <a title="Close sidebar" className={closeBtn} onClick={onCloseClick}>
      ❌
    </a>
    }
  </div>


export default Header
