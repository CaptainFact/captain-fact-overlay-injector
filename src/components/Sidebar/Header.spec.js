import React from 'react'
import Header from './Header'

test('default render', () => {
  snapshot(<Header videoHashId="xxxx" t={tMock} />)
})

test('show toggle if given a close function', () => {
  snapshot(<Header videoHashId="xxxx" onCloseClick={() => 42} t={tMock} />)
})

test('new tab image can be override', () => {
  snapshot(
    <Header videoHashId="xxxx" imgNewTab="overrided-pic.jpg" t={tMock} />
  )
})

test('with empty url params', () => {
  const urlParams = {}
  snapshot(<Header videoHashId="xxxx" urlParams={urlParams} t={tMock} />)
})

test('with statement in url params', () => {
  const urlParams = { statement: 61 }
  snapshot(<Header videoHashId="xxxx" urlParams={urlParams} t={tMock} />)
})
