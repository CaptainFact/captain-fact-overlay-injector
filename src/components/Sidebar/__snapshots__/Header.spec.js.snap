// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render 1`] = `
<div
  className="header"
>
  <a
    href="https://testing.captainfact.io/videos/xxxx?"
    rel="noopener noreferrer"
    target="_BLANK"
    title="Open discussion on CaptainFact"
  >
    <h1
      className="title"
    >
      CaptainFact 
      <img
        alt=""
        src="test-file-stub"
      />
    </h1>
  </a>
</div>
`;

exports[`new tab image can be override 1`] = `
<div
  className="header"
>
  <a
    href="https://testing.captainfact.io/videos/xxxx?"
    rel="noopener noreferrer"
    target="_BLANK"
    title="Open discussion on CaptainFact"
  >
    <h1
      className="title"
    >
      CaptainFact 
      <img
        alt=""
        src="overrided-pic.jpg"
      />
    </h1>
  </a>
</div>
`;

exports[`show toggle if given a close function 1`] = `
<div
  className="header"
>
  <a
    href="https://testing.captainfact.io/videos/xxxx?"
    rel="noopener noreferrer"
    target="_BLANK"
    title="Open discussion on CaptainFact"
  >
    <h1
      className="title"
    >
      CaptainFact 
      <img
        alt=""
        src="test-file-stub"
      />
    </h1>
  </a>
  <button
    className="closeBtn"
    onClick={[Function]}
    title="Close sidebar"
  >
    <img
      alt="Close"
      src="test-file-stub"
    />
  </button>
</div>
`;
